components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -249,113 +246,134 @@ public class LegacyJobOrchestrator : IJobOrchestrator, IPublisherConfigServices,
             return BitConverter.ToString(checksum).Replace("-", string.Empty);
         }
 
+        /// <summary>
+        /// Deserialize string representing published nodes file into PublishedNodesEntryModel entries and
+        /// run schema validation if that is enables.
+        /// </summary>
+        /// <param name="content"></param>
+        /// <returns></returns>
+        private IEnumerable<PublishedNodesEntryModel> DeserializePublishedNodes(string content) {
+            if (!File.Exists(_legacyCliModel.PublishedNodesSchemaFile)) {
+                _logger.Information("Validation schema file {PublishedNodesSchemaFile} does not exist or is disabled, ignoring validation of {publishedNodesFile} file...",
+                _legacyCliModel.PublishedNodesSchemaFile, _legacyCliModel.PublishedNodesFile);
+                return _publishedNodesJobConverter.Read(content, null);
+            }
+            else {
+                using (var fileSchemaReader = new StreamReader(_legacyCliModel.PublishedNodesSchemaFile)) {
```

Comment:
>StreamReader [](http://example.com/codeflow?start=50&length=12)

I guess we should put this in a try catch block and in case of exception we should fallback on converting the content without a schema.

Reply:
Added exception handling for `IOException`s and fallback to deserialization without schema validation.